scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/ParquetCollectionFormatForwardCompatibility.scala

```diff
@@ -0,0 +1,153 @@
+package com.twitter.scalding.parquet.scrooge
+
+import java.util
+
+import org.apache.parquet.schema.Type.Repetition
+import org.apache.parquet.schema.{GroupType, MessageType, Type}
+import org.apache.parquet.thrift.DecodingSchemaMismatchException
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+
+object ParquetCollectionFormatForwardCompatibility {
```

Comment:
Parquet is complicated because there are always 4 schemas involved: the write-time schema, the read-time schema, and a projected schema in terms of the write-schema, and a projected schema in terms of the read-time schema. I find this really hard to keep track of, so I think it'd be a huge help to me and future readers to put a really detailed comment on this class explaining what it's for, what it makes compatible, what the problem is, how it solves it, etc. I think for something like this, the more detailed the better.